src/main/java/bit/javaoop/Superior.java

```diff
@@ -0,0 +1,41 @@
+package bit.javaoop;
+
+import java.util.List;
+
+public interface Superior {
+    List<Inferior> getInferiors();
+
+    void shouldAddEmployee(Inferior employee);
+
+    default void mergeInferiorList(Superior other)
+    {
+        if(other != null && other.getInferiors().size() != 0)
+        {
+            other.getInferiors().stream()
+                    .filter(x -> !getInferiors().contains(x))
+                    .forEach(y -> getInferiors().add(y));
+        }
+    }
+
+    default String printInferiors(int level)
```

Comment:
Trochę nie podoba mi się to wykorzystanie defaultów bo zrobiła nam się z interfejsu normalna klasa po której dziedziczymy aczkolwiek podejrzewam, że to akurat jest kwestia sporna.